### DISCUSSION TITLE
Where to find list of available textures?

### DISCUSSION THREAD

**Discussion by Hedonical** [bjlittle/geovista#990] [Q&A] [answered]
Hi everyone,

First off huge thank you to everyone contributing. I have been testing this package out and have been loving the performance and visuals it can produce. I was looking through the [example code](https://geovista.readthedocs.io/en/latest/tutorials/region-manifold-extraction.html) and specifically have a question about `plotter.add_base_layer(texture=gv.natural_earth_hypsometric())`. 

Is there a location that lists all of the available textures and maybe ways to add your own? Apologies ahead of time if it is in an obvious place, I looked around the GitHub and was unable to find anything.

**Comment by bjlittle** [comment, marked as answer]
Hey @Hedonical!

Thanks so much for the feedback - it's really appreciated indeed :beers: 

So glad that you've found `geovista` and enjoying the experience so far :smile: 

Okay, so yes, there are other textures available "out-the-box" with `geovista`

You can find them squirreled away in the [geovista.pantry.textures](https://geovista.readthedocs.io/en/latest/reference/generated/api/geovista/pantry/textures/index.html) module.

So, for example you can do the following:

```python
import geovista as gv
import geovista.pantry.textures

p = gv.GeoPlotter()
p.add_base_layer(texture=gv.blue_marble())
p.show()
```

As a convenience, these textures are explicitly imported into the `geovista` namespace. Alternatively, the following is explicitly equivalent to the above:

```python
import geovista as gv
import geovista.pantry.textures

p = gv.GeoPlotter()
p.add_base_layer(texture=geovista.pantry.textures.blue_marble())
p.show()
```

Our texture mapping support is rather primitive at the moment, and is only available through the `geovista.geoplotter.GeoPlotterBase.add_base_layer` method, so if you're wanting to do something different, then I'd be interested in your use case.

The deal at the moment is that the texture you provide to `add_base_layer` is used to wrap a full globe, but I have work in the pipeline where you will be able to drape a texture over any geolocated mesh ... so watch out for that landing soon (hopefully :wink:)

In the meantime, you can indeed give any `.png` or `.jpg` image as a texture and `geovista` will wrap it on the base layer.

e.g., rather than the stock NASA Blue Marble, you may alternatively want to use the lush [NASA Black Marble](https://earthobservatory.nasa.gov/images/90008/night-light-maps-open-up-new-applications).

If you manually download the NASE Black marble, say to the file `black-marble.jpg`, you can then do the following:

```python
import geovista as gv
import pyvista as pv

texture = pv.read_texture("blue-marble.jpg")

p = gv.GeoPlotter()
p.add_base_layer(texture=texture)
p.show()
```

Which will give the following:


https://github.com/bjlittle/geovista/assets/2051656/a38e1dd0-232a-46b5-9d3b-c401184e58fc

Does this help?

**Reply by Hedonical** [reply]
Thank you so much! That was an amazing answer and helps a ton.

**Comment by bjlittle** [comment]
@Hedonical You've inspired me to raise issue #992, to add the NASA Black Marble to `geovista` :+1:

**Comment by bjlittle** [comment]
@Hedonical Just out of curiosity, what platform are you running `geovista` on?

**Reply by Hedonical** [reply]
I am running on the latest stable version of Kubuntu with the latest versions of conda and Python. I am away from my machine but could get you more specific details if that would be helpful.

**Reply by bjlittle** [reply]
Thanks @Hedonical !

That's perfect 👍

I was just curious what platform you were using.

I develop on Ubuntu or Red Hat Linux, and it's difficult for me to test hands on with Windows or Mac.

We provided `geovista` on both those platforms but I don't explicitly test them on our GitHub CI, so at the moment rely on them "just working".

I may consider complimenting our test coverage on Ubuntu to run our testing on Win and Mac GitHub Action runners too. But previously when I investigated this it added quite a bit of time to our CI runs.

Something for me to ponder at this end 🤔

Anyways, much obliged, and thanks again 💯
